src/nodes/cannonball.rs

```diff
@@ -0,0 +1,204 @@
+use macroquad::{
+    color,
+    experimental::{
+        animation::{AnimatedSprite, Animation},
+        collections::storage,
+        scene::RefMut,
+    },
+    prelude::*,
+};
+
+use crate::{components::PhysicsBody, Resources};
+
+//use super::EruptedItem;
+
+pub struct Cannonball {
+    cannonball_sprite: AnimatedSprite,
+    body: PhysicsBody,
+    lived: f32,
+    countdown: f32,
+    owner_id: u8,
+    owner_safe_countdown: f32,
+    //   True if erupting from a volcano
+    //erupting: bool,
+    //     When erupting, enable the collider etc. after passing this coordinate on the way down. Set/valid
+    //     only when erupting.
+    //erupting_enable_on_y: Option<f32>,
+}
+
+impl Cannonball {
+    const CANNONBALL_COUNTDOWN_DURATION: f32 = 0.5;
+    /// After shooting, the owner is safe for this amount of time. This is crucial, otherwise, given the
+    /// large hitbox, they will die immediately on shoot.
+    /// The formula is simplified (it doesn't include mount position, run speed and throwback).
+    const CANNONBALL_OWNER_SAFE_TIME: f32 =
+        Self::EXPLOSION_RADIUS / Self::CANNONBALL_INITIAL_SPEED_X_REL;
+
+    const CANNONBALL_WIDTH: f32 = 32.;
+    pub const CANNONBALL_HEIGHT: f32 = 36.;
+    const CANNONBALL_ANIMATION_ROLLING: &'static str = "rolling";
+    const CANNONBALL_INITIAL_SPEED_X_REL: f32 = 600.;
+    const CANNONBALL_INITIAL_SPEED_Y: f32 = -200.;
+    //const CANNONBALL_MOUNT_X_REL: f32 = 20.;
+    //const CANNONBALL_MOUNT_Y: f32 = 40.;
+
+    const EXPLOSION_RADIUS: f32 = 4. * Self::CANNONBALL_WIDTH;
+
+    // Use Cannonball::spawn(), which handles the scene graph.
+    //
+    fn new(pos: Vec2, facing: bool, owner_id: u8) -> Self {
+        // This can be easily turned into a single sprite, rotated via DrawTextureParams.
+        //
+        let mut resources = storage::get_mut::<Resources>();
+        let cannonball_sprite = AnimatedSprite::new(
+            Self::CANNONBALL_WIDTH as u32,
+            Self::CANNONBALL_HEIGHT as u32,
+            &[Animation {
+                name: Self::CANNONBALL_ANIMATION_ROLLING.to_string(),
+                row: 0,
+                frames: 1,
+                fps: 1,
+            }],
+            true,
+        );
+
+        let speed = if facing {
+            vec2(
+                Self::CANNONBALL_INITIAL_SPEED_X_REL,
+                Self::CANNONBALL_INITIAL_SPEED_Y,
+            )
+        } else {
+            vec2(
+                -Self::CANNONBALL_INITIAL_SPEED_X_REL,
+                Self::CANNONBALL_INITIAL_SPEED_Y,
+            )
+        };
+
+        let mut body = PhysicsBody::new(
+            &mut resources.collision_world,
+            pos,
+            0.0,
+            vec2(Self::CANNONBALL_WIDTH, Self::CANNONBALL_HEIGHT),
+        );
+        body.speed = speed;
+
+        /*PhysicsBody {
```

Comment:
Git is pretty OK for keeping older version of the code, I would prefer to avoid commented for historical purposes code